docs/troubleshoot-distributed.md

```diff
@@ -0,0 +1,150 @@
+# Troubleshooting distributed setup
+
+`torchprime` is designed for scaled, distributed training. 
+Once properly configured, the `tp run` command-line function 
+will coordinate several infrastructure tools to move your
+training code onto a cluster, train, and log results. 
+
+However, the trade-off is the investment of time
+getting `tp` configured to properly coordinate with your cluster. 
+
+## tp doctor
+
+To validate that `tp` is configured correctly, run
+
+```sh
+tp doctor
+```
+
+This runs through a series of checks, highlighting additional
+configurations you may need to run.
+
+## End-to-end test
+
+A simple first script to run using `tp run` is the `system_check.py` script. 
+This script will log data about the host, and attempt to run minimal
+calculation with PyTorch using both the CPU and XLA backends. 
+
+```sh
+tp run torchprime/tools/system_check.py --name first-run
```

Comment:
Does the `--name first-run` gets picked up by `tp run` or does it get proxied to the script? I thought I had to do `tp run --name first-run torchprime/tools/system_check.py` but if this works too then great!